### DISCUSSION TITLE
How to modify the pixel through "VIPS_REGION_ADDR"？

### DISCUSSION THREAD

**Discussion by Zhaoyazhou1996** [libvips/libvips#2281] [Q&A] [answered]
Hi，I try to modify the pixel through "VIPS_REGION_ADDR"， it really works in most case, however, some tiff data will report errors. My code is as follow：
```c
    for (int b = 0; b < bands; b++){
                for (int y = 0; y < r.height; y++) {
                    for (int x = 0; x < r.width; x++) {                           
                            VIPS_REGION_ADDR(region, r.left + x, r.top + y)[b] = 0;
                    }  
                }
            }
vips_image_write_to_file(image, filePath, NULL);
    g_object_unref(region);
    g_object_unref(image);
```
 
![error1](https://user-images.githubusercontent.com/49519766/120647796-8f730c00-c4ad-11eb-9e7e-7593d75919e1.png)
![error2](https://user-images.githubusercontent.com/49519766/120647804-90a43900-c4ad-11eb-8b88-47e53ded1f28.png)

I don't know if that's right. Looking forword to your reply! In addition, I‘m very sorry that i can't donate libvips through paypal or credit card in China. ┬_┬

**Comment by jcupitt** [comment, marked as answer]
Hi again @Zhaoyazhou1996,

You need to call `vips_image_inplace()` before making in-place modifications to an image:

https://libvips.github.io/libvips/API/current/VipsImage.html#vips-image-inplace

This is how the `draw` operations work:

https://libvips.github.io/libvips/API/current/libvips-draw.html

**Reply by Zhaoyazhou1996** [reply]
Oh! it's really work! Thanks a lot. :)
